src/Store/CookieStore.php

```diff
@@ -26,231 +26,257 @@ class CookieStore implements StoreInterface
      */
-    private string $cookiePrefix;
+    private string $namespace;
 
     /**
-     * Number of bytes to deduct/buffer from KEY_CHUNKING_THRESHOLD before chunking begins.
+     * The threshold (in bytes) in which chunking/splitting occurs.
      */
-    private int $chunkingThreshold;
+    private int $threshold;
+
+    /**
+     * Internal cache of the storage state.
+     *
+     * @var array<mixed>
+     */
+    private array $store = [];
+
+    /**
+     * When true, CookieStore will not setState() itself. You will need manually call the method to persist state to storage.
+     */
+    private bool $deferring = false;
 
     /**
      * CookieStore constructor.
      *
      * @param SdkConfiguration $configuration   Base configuration options for the SDK. See the SdkConfiguration class constructor for options.
-     * @param string           $cookiePrefix    A string to prefix stored cookie keys with.
+     * @param string           $namespace       A string in which to store cookies under on devices.
      */
     public function __construct(
         SdkConfiguration $configuration,
-        string $cookiePrefix = 'auth0'
+        string $namespace = 'auth0'
     ) {
-        [$cookiePrefix] = Toolkit::filter([$cookiePrefix])->string()->trim();
+        [$namespace] = Toolkit::filter([$namespace])->string()->trim();
 
         Toolkit::assert([
-            [$cookiePrefix, \Auth0\SDK\Exception\ArgumentException::missing('cookiePrefix')],
+            [$namespace, \Auth0\SDK\Exception\ArgumentException::missing('namespace')],
         ])->isString();
 
         $this->configuration = $configuration;
-        $this->cookiePrefix = $cookiePrefix ?? 'auth0';
+        $this->namespace = hash(self::KEY_HASHING_ALGO, $namespace ?? 'auth0');
+        $this->threshold = self::KEY_CHUNKING_THRESHOLD - strlen($this->namespace) + 2;
 
-        $this->chunkingThreshold = self::KEY_CHUNKING_THRESHOLD - strlen(hash(self::KEY_HASHING_ALGO, 'threshold'));
+        $this->getState();
     }
 
     /**
-     * Persists $value on cookies, identified by $key.
-     *
-     * @param string $key   Cookie to set.
-     * @param mixed  $value Value to use.
+     * Returns the current namespace identifier.
      */
-    public function set(
-        string $key,
-        $value
-    ): void {
-        $key = $this->getCookieName($key);
-        $cookieOptions = $this->getCookieOptions();
-
-        $value = $this->encrypt($value);
-
-        $_COOKIE[$key] = $value;
-
-        if (strlen($value) >= $this->chunkingThreshold) {
-            // @phpstan-ignore-next-line
-            $chunks = str_split($value, $this->chunkingThreshold);
-
-            // @phpstan-ignore-next-line
-            if ($chunks !== false) {
-                $chunkIndex = 1;
-
-                setcookie(join(self::KEY_SEPARATOR, [ $key, '0']), (string) count($chunks), $cookieOptions);
+    public function getNamespace(): string
```

Comment:
This is also never used. I suggest removing it.

Reply:
This is used by `CookieStoreTest` for unit testing purposes